docs/monorepo/history_strategy.md

```diff
@@ -0,0 +1,218 @@
+# La Grande Fusion with History
+
+**Goal**: retain valuable history in a way that is consumable in the final merge layout.
+
```

Comment:
Some more intro could be helpful here for someone reading this in 6-12 months. Maybe something like, "These are the steps we followed to selectively retain flutter/engine repo history when merging the engine repo into the flutter/flutter repo. The idea was to retain as much useful history as possible without blowing up the footprint of the framework's .git folder."